### PR TITLE
[UPDATED] boolean for tuning added

### PR DESCRIPTION
**Pull request by puiyanng**

### CODE REVIEW

File: modelling.py [hunk 1]
```diff
@@ -52,12 +55,40 @@ def fit_random_forest(features, labels):
...
     return model
 
+def fit_KNN(features, labels):
+    model = KNeighborsClassifier(n_neighbors=14)
+    model.fit(features, labels)
+
+    return model
+
+def fit_gradient_boosting(features, labels):
+    """
```
Comment (DanishAlsayed) [inline modelling.py]: Why is this commented?

File: modelling.py [hunk 2]
```diff
@@ -52,12 +55,40 @@ def fit_random_forest(features, labels):
...
+        estimator=GradientBoostingClassifier(min_samples_split=2,
+                                             min_samples_leaf=1, subsample=1, max_features='sqrt', random_state=10),
+        param_grid=param_grid, scoring='accuracy', n_jobs=4, iid=False, cv=5)
+    
+    
+    model.fit(features, labels)
+    print(model.cv_results_, model.best_params_, model.best_score_)
+    """
+
+    model = GradientBoostingClassifier(learning_rate=0.001, n_estimators=500,max_depth=3, min_samples_split=2, min_samples_leaf=2, subsample=1,max_features='sqrt', random_state=10)
```
Comment (DanishAlsayed) [inline modelling.py]: How did you decide on these parameters? I thought you will find them via hyperparameter tuning/search?

File: results.csv [hunk 3]
```diff
@@ -1,4 +1,5 @@
 Model,%age correct,feature excluded,comments
 AdaBoost Classifier,51.6,Date,"Preliminary, 80/20 randomly sliced"
 Random Forest Classifier,51.2,Date,"Preliminary, 80/20 randomly sliced"
-SVM, 51.2,Date,"Preliminary, 80/20 first 80% for training and last 20% for testing"
\ No newline at end of file
+SVM, 51.2,Date,"Preliminary, 80/20 first 80% for training and last 20% for testing"
+KNN,57.7,Date,"Preliminary, 80/20 randomly sliced"
```
Comment (DanishAlsayed) [inline results.csv]: Nice

File: feature_engineering.py [hunk 4]
```diff
@@ -126,12 +136,23 @@ def data_integrate():
     # data_1d_5y.head(5)
     data_1d_5y[data_1d_5y.isna().any(axis=1)]
     data_1d_5y.reset_index(inplace=True)
-    data_1d_5y.to_csv('data_id_5y.csv', index=True) # Save the name as 'data_id_5y.csv'
+    data_1d_5y.to_csv('data_1d_5y.csv', index=True) # Save the name as 'data_1d_5y.csv'
+
+def generate_data_reg():
```
Comment (DanishAlsayed) [inline feature_engineering.py]: Not sure why you need this method?

File: feature_engineering.py [hunk 5]
```diff
@@ -126,12 +136,23 @@ def data_integrate():
...
+    data = pd.read_csv('data/data_1d_5y.csv')
+    data_norm = pd.read_csv('data/data_normalized_1d_5y.csv')
+
+    data_norm = data_norm.drop(["Signal"], axis=1)
+    data_norm['Y'] = generate_y_reg(data, 'Close').shift(-1)
+    data_norm = data_norm[
+        ['Date', 'Y', 'Open', 'High', 'Low', 'Close', 'Volume', 'RUSS_Close', 'NYSE_Close', 'NASDAQ_Close', 'DJI_Close',
```
Comment (DanishAlsayed) [inline feature_engineering.py]: Too much hardcoding. What if we decide to add a feature? Then we will have to add its name explicitely here too. Please loop over the column names programmatically

File: data/data_reg_1d_5y.csv [hunk 6]
```diff
@@ -0,0 +1,1240 @@
+Date,Y,Open,High,Low,Close,Volume,RUSS_Close,NYSE_Close,NASDAQ_Close,DJI_Close,FTSE_Close,GOLD_Close,N225_Close,USDCNY_Close,USDEUR_Close,USDGBP_Close,USDJPY_Close
```
Comment (DanishAlsayed) [inline data/data_reg_1d_5y.csv]: Did you verify that the values are correct? I usually do that by randomly picking a few cells and calculating the value that should be there manually to check if they match.

File: modelling.py [hunk 7]
```diff
@@ -52,12 +55,40 @@ def fit_random_forest(features, labels):
...
-    grid_search = GridSearchCV(svm, param_grid, n_jobs = 8, verbose=1)    
-    model = grid_search.fit(features, labels) 
+    svm = SVC(kernel='rbf', probability=True)
+    param_grid = {'C': [1e-3, 1e-2, 1e-1, 1, 10, 100, 1000], 'gamma': [0.001, 0.0001]}
+    grid_search = GridSearchCV(svm, param_grid, n_jobs = 8, verbose=1)
+    model = grid_search.fit(features, labels)
     return model
 
+def fit_KNN(features, labels):
+    model = KNeighborsClassifier(n_neighbors=14)
```
Comment (DanishAlsayed) [inline modelling.py]: No hyper parameter tuning/search?

File: results.csv [hunk 8]
```diff
@@ -1,4 +1,5 @@
 Model,%age correct,feature excluded,comments
```
Comment (DanishAlsayed) [inline results.csv]: Where are the results from the gradient boosting model?

File: feature_engineering.py (hunk 4 above)
Comment (puiyanng) [reply feature_engineering.py]: I'm just correcting the typo haha

File: modelling.py (hunk 1 above)
Comment (puiyanng) [reply modelling.py]: I am commenting the hyperparameter tuning cuz it takes a long time to run

File: modelling.py (hunk 2 above)
Comment (puiyanng) [reply modelling.py]: yup, the commented part

File: modelling.py (hunk 7 above)
Comment (puiyanng) [reply modelling.py]: I used a for loop
